dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -96,7 +96,7 @@ public AbstractRegistry(URL url) {
 
     protected static List<URL> filterEmpty(URL url, List<URL> urls) {
         if (urls == null || urls.isEmpty()) {
-            List<URL> result = new ArrayList<URL>(1);
+            List<URL> result = new ArrayList<>(1);
             result.add(url.setProtocol(Constants.EMPTY_PROTOCOL));

```

Comment:
curios to know why we to change the protocol type?

Reply:
Yes I know that, I didn't even change line 100, and the git diff show nothing changed in this line

Reply:
No problem.